library/include/MiscUtils.h

```diff
@@ -310,6 +310,27 @@ Link *linked_list_insert_after(Link *pos, Link *link)
     return link;
 }
 
+/**
+ * Returns true if the item with id idToRemove was found, deleted, and removed
+ * from the list. Otherwise returns false.
+ */
+template<typename Link>
+bool linked_list_remove(Link *head, int32_t idToRemove)
```

Comment:
Ah, I didn't consider that this is difficult to implement generically. I think it's fine to limit it to the `id` field for now (given that the sole caller works with that) - if we need it to be more generic, the pointer-to-member approach used in [this overload of linear_index()](https://github.com/DFHack/dfhack/blob/5750bfff9b2275733df207f037bd8a04e2c2139f/library/include/MiscUtils.h#L160-L167) could work.